### PR TITLE
Updated documentation for CPU support

### PR DESCRIPTION
**Pull request by aayushg55**
Deleted old `Limitations` page and added new `Executor Compatibility` page

### CODE REVIEW

File: docs_input/api/stats/corr/cov.rst [hunk 1]
```diff
@@ -5,6 +5,10 @@ cov
 
 Compute a covariance matrix
 
+.. note::
+   This function is currently not supported with host-based executors (CPU)
```
Comment (cliffburdick) [inline docs_input/api/stats/corr/cov.rst:9]: Looking at the `cov` code it seems like it should be supported on the host, but just hasn't been updated yet. I don't see any functions that would prevent it from working.

Comment (aayushg55) [reply docs_input/api/stats/corr/cov.rst:9]: Yea, we just need to change the streams to executors

**Comment by cliffburdick** [comment]
Overriding the failed CI since these are docs changes and the docs building passed.

**Pull request merged**